components/builder-depot/src/server.rs

```diff
@@ -784,7 +791,16 @@ fn upload_package(req: &mut Request) -> IronResult<Response> {
     let temp_name = format!("{}.tmp", Uuid::new_v4());
     let temp_path = parent_path.join(temp_name);
 
-    let mut archive = write_archive(&temp_path, &mut req.body)?;
+    let mut archive = match write_archive(&temp_path, &mut req.body) {
+        Ok(a) => a,
+        Err(_) => {
+            return Ok(Response::with((
+                status::InternalServerError,
+                "Internal Habitat error",
```

Comment:
Minor - use ideomatic format "xxx:yyy:NNN" for the error string?

Reply:
Sure, I can change that.  FWIW, I didn't write any of this error code.  I just moved it from the bottom of the file to here, but your suggestion is a good one.